.gitlab/kernel_version_testing/security_agent.yml

```diff
@@ -0,0 +1,235 @@
+    paths:
+      - $CI_PROJECT_DIR/connector-${ARCH}
+
+upload_security_agent_tests_x64:
+  extends:
+    - .upload_security_agent_tests
+  needs: ["go_deps", "prepare_ebpf_functional_tests_x64", "tests_ebpf_x64"]
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_x64$DATADOG_AGENT_SYSPROBE_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
+  tags: ["arch:amd64"]
+  variables:
+    ARCH: x86_64
+    INSTANCE_TYPE: m5d.metal
+
+upload_security_agent_tests_arm64:
+  extends:
+    - .upload_security_agent_tests
+  needs: ["go_deps", "prepare_ebpf_functional_tests_arm64", "tests_ebpf_arm64"]
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_arm64$DATADOG_AGENT_SYSPROBE_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
+  tags: ["arch:arm64"]
+  variables:
+    ARCH: arm64
+    INSTANCE_TYPE: m6gd.metal
+
+.kernel_matrix_testing_run_security_agent_tests:
+  extends: .kernel_matrix_testing_run_tests
+  stage: kernel_matrix_testing_security_agent
+  rules: !reference [.on_security_agent_changes_or_manual]
+  variables:
+    TEST_COMPONENT: security-agent
+
+kernel_matrix_testing_run_security_agent_tests_x64:
+  extends:
+    - .kernel_matrix_testing_run_security_agent_tests
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_x64$DATADOG_AGENT_SYSPROBE_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
+  tags: ["arch:amd64"]
+  needs:
+    [
+      "kernel_matrix_testing_setup_env_security_agent_x64",
+      "upload_dependencies_security_agent_x64",
+    ]
+  timeout: 3h
+  variables:
+    ARCH: "x86_64"
+  parallel:
+    matrix:
+      - TAG:
+          [
+            "ubuntu_18.04",
+            "ubuntu_20.04",
+            "ubuntu_22.04",
+            "ubuntu_23.10",
+            "amzn_5.4",
+            "amzn_5.10",
+            "fedora_37",
+            "fedora_38",
+            "debian_10",
+            "debian_11",
+            "debian_12",
+            "centos_79",
+          ]
+        TEST_SET: [all_tests]
+
+kernel_matrix_testing_run_security_agent_tests_arm64:
+  extends:
+    - .kernel_matrix_testing_run_security_agent_tests
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_arm64$DATADOG_AGENT_SYSPROBE_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
+  tags: ["arch:arm64"]
+  needs:
+    [
+      "kernel_matrix_testing_setup_env_security_agent_arm64",
+      "upload_dependencies_security_agent_arm64",
+    ]
+  timeout: 3h
+  variables:
+    ARCH: "arm64"
+  parallel:
+    matrix:
+      - TAG:
+          [
+            "ubuntu_18.04",
+            "ubuntu_20.04",
+            "ubuntu_22.04",
+            "ubuntu_23.10",
+            "amzn_5.4",
+            "amzn_5.10",
+            "fedora_37",
+            "fedora_38",
+            "debian_10",
+            "debian_11",
+            "debian_12",
+            "centos_79",
```

Comment:
How does this list of kernels compare to what security-agent is testing with now? What is missing/added?

Reply:
Here's the list: 

Removed: 

- Amazon 5.15, 4.14 (which [doesn't seem to advertised as supported](https://docs.datadoghq.com/security/cloud_security_management/setup/?tab=csmworkloadsecurity#prerequisites)) (x86)

Removed, although the jobs were marked as manual in kitchen tests:
- Centos 7.7 
- RHEL 8.5, 9.1
- SLES 12, 15 
- OpenSUSE 15.3
- Oracle 7.9 
- Rocky Linux 9.2

Added: 
 - All ARM architectures except for Ubuntu 20, 22 which were already being tested
 - Debian 12 
 - Fedora 37, 38

Reply:
Can we please add this in the description?

Reply:
added